src/rust/bitbox02-rust-c/src/util.rs

```diff
@@ -121,22 +123,35 @@ impl AsRef<str> for CStrMut {
             buf = core::ptr::NonNull::dangling().as_ptr();
         }
         assert!(!buf.is_null());
-        unsafe { core::str::from_utf8_unchecked(core::slice::from_raw_parts(buf, self.len)) }
+
+        // Find null terminator
+        let mut len = 0;
+        unsafe {
+            let mut b = buf;
+            while b.read() != 0 {
+                b = b.offset(1);
+                len += 1;
+                if len == self.len {
+                    panic!("CStrMut not null terminated");
+                }
+            }
+        }
+
+        unsafe { core::str::from_utf8_unchecked(core::slice::from_raw_parts(buf, len)) }
```

Comment:
actually, if we want to panic on non-utf8 bytes we can remove the `unsafe` here and use `core::str::from_utf8()` (which isn't unsafe, because it checks that the string is a valid utf-8 encoded string).

Reply:
`from_raw_parts` is unsafe though?